test/compound_assignment_test.js

```diff
@@ -29,6 +29,33 @@ describe('compound assignment', () => {
     check(`a ^= 1`, `a ^= 1;`);
   });
 
+  it('handles compound modulo assignment', () => {
+    check(`
+      a %%= b
+    `, `
+      let a = __mod__(a, b);
```

Comment:
This seems like an issue with add-variable-declarations, and possibly also esnext. Assignment to a binding that is also referenced in the rvalue should probably assume a global. Also, even if there is such a `var`, turning it into a `let` introduces a TDZ issue.